### DISCUSSION TITLE
"Display persona card on hover" doesn't work

### DISCUSSION THREAD

**Discussion by jochenfritz-22** [microsoft-search/pnp-modern-search#3343] [Q&A] [answered]
Hello, I want to set up a user directory for my company and I'm currently having the following problem:
I can display the users and style the cards, which works well. Unfortunately I can't figure out how to activate the person card on hover. I read all the posts but couldn't find a solution. Could you kindly help me to solve this problem?

I have the following configuration:

- Version: PnP - Modern Search 4.9.2.0

- Language: German

- Availability: All sites

- API Access: All required scopes approved

- Source PnP Search Result: SahrePoint

- Show Persona Card on hover: Is enabled
![Bedingung 03](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/f5e7c2d5-971b-4f7e-9235-b2b6099f920c)

- Mapping: The PersonQuery slot is mapped to UserName and the UserEmail is also mapped to the UserName
`<mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">`
![Bedingung 02](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/478011bd-f359-4b20-94d8-feaa0172dd7a)

Where exactly is the error, what am I missing?
Thank you in advance for supporting me

**Comment by kasperbolarsen** [comment]
Hi @jochenfritz-22 

Does it work when using the out of the box People layout?

I noted that the out of the box has PersonQuery pointing at AADObjectId:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/02b43e4f-369d-4343-a1c7-c38cf0c9d444)

**Comment by jochenfritz-22** [comment]
Hello @kasperbolarsen, Thanks for your feedback, when I reset the value to AADObjectId no result is found.

![Bedingung 04](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/c7f7d474-90f6-448b-baf3-61f31321fcc2)

How was this originally intended, what values do you have to use? In the ticket #1772 describes @wobba:
Looking at the latest version the template uses <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover"> where the PersonQuery slot is mapped to UserName. So it should in theory be set up correct. UserEmail is also mapped to the UserName managed property by default.

**Reply by kasperbolarsen** [reply]
It should work out of the box. Add a new page, add the results web part, select SharePoint,  the query to *, change the source to peoplesearch, select the people layout,  enable hover and publish. 
You should not have to change anything in the slots.

**Reply by wobba** [reply]
Ensure the person is a native and not a guest user as well.

**Reply by jochenfritz-22** [reply]
Hello @kasperbolarsen and @wobba, If I follow all the steps as described by you, I no longer get any results when I activate the hover slider:
![Bildschirmfoto 2023-11-20 um 18 46 30](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/add57e55-2915-4c6f-9723-f6c493bd3d74)
![Bildschirmfoto 2023-11-20 um 18 46 17](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/5e8d294f-6ab6-46d6-b84f-5d5733059b38)
![Bildschirmfoto 2023-11-20 um 18 43 17](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/9edd705f-8761-4d68-b213-ac7daff574c8)
I also tried it with a single user who is clearly not a guest to avoid the results containing guest profiles. Unfortunately no difference.

**Reply by kasperbolarsen** [reply]
That is weird. Try replacing the query you own email,  and perhaps change the layout to debug in order to inspect the data

**Reply by jochenfritz-22** [reply]
Hello @kasperbolarsen, In debug mode, "AADObjectID" and "UserName" have the values I would expect. Or am I missing something?...
![Bildschirmfoto 2023-11-20 um 19 29 51](https://github.com/microsoft-search/pnp-modern-search/assets/122534001/7ad58950-05a7-4da8-b556-a0ee7063fb3c)

**Reply by whitehotaru** [reply]
Sorry to chime in, but I had a similar problem. Could it be that you are using a custom display template? For us, everything worked as described with the standard display template, but as soon as we switched to custom, there were no more results (and no more people card options as well).

**Reply by jochenfritz-22** [reply]
Hello @whitehotaru, If I am using the WebPart out of the box without any customizing, I don't get the persona card on hover.

**Comment by kasperbolarsen** [comment]
If possible I will recommend to delete the web parts from the app catalog,  and install 4.9.3. It sounds like the app permissions didn't take.

**Reply by jochenfritz-22** [reply]
Hello @kasperbolarsen, I have updated from version 4.9.2 to version 4.9.3. However, that didn't solve the problem.
But what you mean, is to properly delete the APP from the APP catalog and then reinstall it. Correct? I'm wondering if I have to re-approve the required scopes after reinstallation or if the last settings will be reapplied?

**Reply by kasperbolarsen** [reply, marked as answer]
I have a suspicion that the permissions somehow failed to be granted,  so delete and install again in the app catalog might cause it to work as expected

**Reply by jochenfritz-22** [reply]
Hello @kasperbolarsen, thank you for your support. Sometimes the solutions are more obvious than expected. I deleted the APP in the APP Catalog and installed a new version. The requested scope for “User.Read.All” then had to be approved again. This allowed the Persona Card to be displayed on hover.

**Reply by kasperbolarsen** [reply]
Excellent!!
